webservices/resources/candidate_aggregates.py

```diff
@@ -240,61 +240,78 @@ def build_query(self, **kwargs):
-        check_cand_status = (
-            db.session.query(history)
-            .filter(
-                history.candidate_id == total.candidate_id,
-                history.candidate_election_year == total.election_year,
-                history.candidate_inactive.is_(True),
-            )
-            .exists()
-        )
-
         query = db.session.query(
-            sa.func.substr(total.candidate_id, 1, 1).label('office'),
+            total.office.label('office'),
             total.election_year.label('election_year'),
-            sa.func.sum(total.receipts).label('total_receipts'),
-            sa.func.sum(total.disbursements).label('total_disbursements'),
-        ).filter(
-            total.is_election == True  # noqa
+            sa.func.sum(total.receipts).label(
+                'total_receipts'),
+            sa.func.sum(total.disbursements).label(
+                'total_disbursements'),
+            sa.func.sum(total.individual_itemized_contributions).label(
+                'total_individual_itemized_contributions'),
+            sa.func.sum(total.transfers_from_other_authorized_committee).label(
+                'total_transfers_from_other_authorized_committee'),
+            sa.func.sum(total.other_political_committee_contributions).label(
+                'total_other_political_committee_contributions'),
         )
+        # remove election_year=null result
+        query = query.filter(~total.election_year.is_(None))
 
         if kwargs.get('office') and kwargs['office'] is not None:
+            query = query.filter(total.office == kwargs['office'])
+
+        if kwargs.get('election_year'):
             query = query.filter(
-                sa.func.substr(total.candidate_id, 1, 1) == kwargs['office']
+                total.election_year.in_(kwargs['election_year'])
             )
-        if kwargs.get('election_year'):
-            query = query.filter(total.election_year.in_(kwargs['election_year']))
 
         if 'is_active_candidate' in kwargs and kwargs.get('is_active_candidate'):
-            query = query.filter(~check_cand_status)
+            query = query.filter(total.candidate_inactive.is_(False))
+
         elif 'is_active_candidate' in kwargs and not kwargs.get('is_active_candidate'):
-            query = query.filter(check_cand_status)
-        else:  # load all candidates
-            pass
+            query = query.filter(total.candidate_inactive.is_(True))
+
+        # if not pass election_full variable, election_full default set `true` in args.py
+        # if pass election_full = true, election_year is candidate election year
+        # if pass election_full = flase, election_year is finance two-year period
+        if 'election_full' in kwargs and kwargs.get('election_full'):
+            query = query.filter(total.is_election.is_(kwargs['election_full']))
+
+        if kwargs.get('min_election_cycle') and kwargs['min_election_cycle'] is not None:
```

Comment:
because `.get()` returns None if the key isn't found, can this be replaced (and in other places) with 

`if kwargs.get('min_election_cycle'):`